### PR TITLE
Fix linting when creating a pipeline skipping some parts of the template

### LINKED ISSUE #2329: nf-core lint fails on newly created pipeline
### Description of the bug

Using the latest version of `dev` when I create a new pipeline with a custom `Pipeline prefix` and `Skip template areas` and then run `nf-core lint` in the pipeline folder the linting fails which shouldn't be the case because it's a fresh pipeline.

Also seeing a number of linting tests that are ignored which don't make any sense. Need to track this down and fix it.

```
$ nf-core create

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 2.9.dev0 - https://nf-co.re


? Workflow name test
? Description test
? Author test
? Do you want to customize which parts of the template are used? Yes
? Pipeline prefix bug
? Skip template areas? [iGenomes config]
INFO     Creating new nf-core pipeline: 'bug/test'                                                                                                                      
INFO     Initialising pipeline git repository                                                                                                                           
INFO     Done. Remember to add a remote and push to GitHub:                                                                                                             
          cd /home/harshil/webinar/bug-test                                                                                                                             
          git remote add origin git@github.com:USERNAME/REPO_NAME.git                                                                                                   
          git push --all origin                                                                                                                                         
INFO     This will also push your newly created dev branch and the TEMPLATE branch for syncing.                                                                         

$ cd bug-test  
$ nf-core lint  

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 2.9.dev0 - https://nf-co.re


INFO     Testing pipeline: .                                                                                                                                            

╭─ [?] 15 Pipeline Tests Ignored ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                                                                                      │
│ files_exist: File is ignored: CODE_OF_CONDUCT.md                                                                                                                     │
│ files_exist: File is ignored: assets/nf-core-test_logo_light.png                                                                                                     │
│ files_exist: File is ignored: docs/images/nf-core-test_logo_light.png                                                                                                │
│ files_exist: File is ignored: docs/images/nf-core-test_logo_dark.png                                                                                                 │
│ files_exist: File is ignored: .github/ISSUE_TEMPLATE/config.yml                                                                                                      │
│ files_exist: File is ignored: .github/workflows/awstest.yml                                                                                                          │
│ files_exist: File is ignored: .github/workflows/awsfulltest.yml                                                                                                      │
│ nextflow_config: Config variable ignored: manifest.name                                                                                                              │
│ nextflow_config: Config variable ignored: manifest.homePage                                                                                                          │
│ files_unchanged: File does not exist: CODE_OF_CONDUCT.md                                                                                                             │
│ files_unchanged: File does not exist: .github/ISSUE_TEMPLATE/config.yml                                                                                              │
│ files_unchanged: File does not exist: assets/nf-core-test_logo_light.png                                                                                             │
│ files_unchanged: File does not exist: docs/images/nf-core-test_logo_light.png                                                                                        │
│ files_unchanged: File does not exist: docs/images/nf-core-test_logo_dark.png                                                                                         │
│ actions_awstest: 'awstest.yml' workflow not found: ./.github/workflows/awstest.yml                                                                                   │
│                                                                                                                                                                      │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ [!] 22 Pipeline Test Warnings ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                                                                                      │
│ files_exist: File not found: conf/igenomes.config                                                                                                                    │
│ readme: README contains the placeholder zenodo.XXXXXXX. This should be replaced with the zenodo doi (after the first release).                                       │
│ pipeline_todos: TODO string in README.md: TODO nf-core:                                                                                                              │
│ pipeline_todos: TODO string in README.md: Include a figure that guides the user through the major workflow steps. Many nf-core                                       │
│ pipeline_todos: TODO string in README.md: Fill in short bullet-pointed list of the default steps in the pipeline                                                     │
│ pipeline_todos: TODO string in README.md: Describe the minimum required steps to execute the pipeline, e.g. how to prepare samplesheets.                             │
│ pipeline_todos: TODO string in README.md: update the following command to include all required parameters for a minimal example                                      │
│ pipeline_todos: TODO string in README.md: If applicable, make list of people who have also contributed                                                               │
│ pipeline_todos: TODO string in README.md: Add citation for pipeline after first release. Uncomment lines below and update Zenodo doi and badge at the top of this    │
│ file.                                                                                                                                                                │
│ pipeline_todos: TODO string in README.md: Add bibliography of tools and data used in your pipeline                                                                   │
│ pipeline_todos: TODO string in nextflow.config: Specify your pipeline's command line flags                                                                           │
│ pipeline_todos: TODO string in test.config: Specify the paths to your test data on nf-core/test-datasets                                                             │
│ pipeline_todos: TODO string in test.config: Give any required params for the test so that command line flags are not needed                                          │
│ pipeline_todos: TODO string in test_full.config: Specify the paths to your full test data ( on nf-core/test-datasets or directly in repositories, e.g. SRA)          │
│ pipeline_todos: TODO string in test_full.config: Give any required params for the test so that command line flags are not needed                                     │
│ pipeline_todos: TODO string in base.config: Check the defaults for all processes                                                                                     │
│ pipeline_todos: TODO string in base.config: Customise requirements for specific processes.                                                                           │
│ pipeline_todos: TODO string in methods_description_template.yml: #Update the HTML below to your prefered methods description, e.g. add publication citation for this │
│ pipeline                                                                                                                                                             │
│ pipeline_todos: TODO string in output.md: Write this documentation describing your workflow's output                                                                 │
│ pipeline_todos: TODO string in usage.md: Add documentation about anything specific to running your pipeline. For general topics, please point to (and add to) the    │
│ main nf-core website.                                                                                                                                                │
│ pipeline_todos: TODO string in WorkflowMain.groovy: Add Zenodo DOI for pipeline after first release                                                                  │
│ pipeline_todos: TODO string in ci.yml: You can customise CI pipeline run tests as required                                                                           │
│                                                                                                                                                                      │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ [✗] 1 Pipeline Test Failed ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                                                                                      │
│ files_unchanged: .github/ISSUE_TEMPLATE/bug_report.yml does not match the template                                                                                   │
│                                                                                                                                                                      │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

╭─ [!] 3 Module Test Warnings ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                           ╷                                             ╷                                                                            │
│ Module name                               │ File path                                   │ Test message                                                               │
│╶──────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────────────────────────────────────────────────╴│
│ custom/dumpsoftwareversions               │ modules/nf-core/custom/dumpsoftwareversions │ New version available                                                      │
│ fastqc                                    │ modules/nf-core/fastqc                      │ New version available                                                      │
│ multiqc                                   │ modules/nf-core/multiqc                     │ New version available                                                      │
│                                           ╵                                             ╵                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭───────────────────────╮
│ LINT RESULTS SUMMARY  │
├───────────────────────┤
│ [✔] 156 Tests Passed  │
│ [?]  15 Tests Ignored │
│ [!]  25 Test Warnings │
│ [✗]   1 Test Failed   │
╰───────────────────────╯

Tip: Some of these linting errors can automatically be resolved with the following command:

    nf-core lint   --fix files_unchanged
```

### PR DESCRIPTION
**Pull request by mirpedrol**
Fix linting when createing a pipeline skipping some parts of the template

Close #2329 

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/create.py [hunk 1]
```diff
@@ -446,6 +469,10 @@ def fix_linting(self):
         if not self.template_params["github_badges"] or not self.template_params["github"]:
             lint_config["readme"] = ["nextflow_badge"]
 
+        # If the pipeline is unbranded
+        if not self.template_params["branded"]:
+            lint_config["files_unchanged"].extend([".github/ISSUE_TEMPLATE/bug_report.yml"])
```
Comment (ewels) [inline nf_core/create.py:474]: If we can avoid just flat out ignoring this file, that would be nice. There is already a [function](https://github.com/nf-core/tools/blob/0912990a63ef29e44e07cc2ba6ab81113684e0ae/nf_core/create.py#L371-L387) to customise it to remove some nf-core specific stuf. I think the failure on this file is something to do with the custom prefix / name match at the end of the output.

Comment (mirpedrol) [reply nf_core/create.py:474]: When the pipeline is unbranded we remove one chunk of this file containing links to nf-core documentation:
```yaml
  - type: markdown
    attributes:
      value: |
        Before you post this issue, please check the documentation:

        - [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
        - [{{ name }} pipeline documentation](https://nf-co.re/{{ short_name }}/usage)
```
so it won't match the template, and can't be considered a partial match either.

**Review by drpatelh — APPROVED** [review]
Tested your branch and the `linting` is now working for me @mirpedrol . Will approve and wait for @ewels to comment on his comment.

File: nf_core/create.py (hunk 1 above)
Comment (drpatelh) [reply nf_core/create.py:475]: > so it won't match the template, and can't be considered a partial match either.

Seems like this isn't possible so will merge this in and we can follow up in a separate PR / issue if required.

**Pull request merged**